middleman_protocol/middleman_protocol/helpers.py

```diff
@@ -0,0 +1,18 @@
+from asyncio import StreamReader
+
+from middleman_protocol.constants import ErrorCode
+from middleman_protocol.constants import FRAME_SEPARATOR
+from middleman_protocol.constants import MIDDLEMAN_EXCEPTION_TO_ERROR_CODE_MAP
+from middleman_protocol.exceptions import MiddlemanProtocolError
+from middleman_protocol.stream import escape_decode_raw_message
+
+
+async def handle_frame_receive_async(reader: StreamReader) -> bytes:
+    raw_data = await reader.readuntil(FRAME_SEPARATOR)
+    index = raw_data.index(FRAME_SEPARATOR)
+    raw_data_without_separator = raw_data[:index]
+    return escape_decode_raw_message(raw_data_without_separator)
```

Comment:
`escape_decode_raw_message` might raise `BrokenEscapingInFrameMiddlemanProtocolError`, please catch it and handle (by returning `None`).

Reply:
Similar to above comment, @dybi suggested to add such error handlers to `middleman/middleman_server.py` where this function is being called?

Reply:
But then it will be mismatch in functionality. Sync version catches this error and returns `None`.

Reply:
@rwrzesien, it will be not a mismatch, but just lack of a hack ;) `None` is not needed in asnync version as we can safely raise exceptions (whereas in sync version raising exceptions would kill generator)